### PR TITLE
Add wrapper class of job scheduler lock service

### PR DESCRIPTION
**Pull request by heemin32**
### Description
Add wrapper class of job scheduler lock service
 
### Issues Resolved
N/A
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/geospatial/ip2geo/common/Ip2GeoLockService.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.jobscheduler.spi.LockModel;
+import org.opensearch.jobscheduler.spi.utils.LockService;
+
+/**
+ * A wrapper of job scheduler's lock service for datasource
+ */
+public class Ip2GeoLockService {
```
Comment (martin-gaievski) [inline src/main/java/org/opensearch/geospatial/ip2geo/common/Ip2GeoLockService.java:28]: Are you planning to add an interface for the service that this class will implement?

**Review by martin-gaievski — COMMENTED** [review]
What's the client for the new service, do you plan another PR for this?

**Comment by heemin32** [comment]
> What's the client for the new service, do you plan another PR for this?

Yes. Didn't include the client part so that each PR can be small.

File: src/main/java/org/opensearch/geospatial/ip2geo/common/Ip2GeoLockService.java (hunk 1 above)
Comment (heemin32) [reply src/main/java/org/opensearch/geospatial/ip2geo/common/Ip2GeoLockService.java:28]: I don't have such plan.

File: src/main/java/org/opensearch/geospatial/ip2geo/common/Ip2GeoLockService.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    private final ClusterService clusterService;
+    private final Client client;
+    private final LockService lockService;
+
+    /**
+     * Default constructor
```
[the remark this answers is not in the document]
Comment (heemin32) [reply src/main/java/org/opensearch/geospatial/ip2geo/common/Ip2GeoLockService.java:34]: If there is constructor with argument, compiler does not create no argument constructor by default. Therefore, we don't need to make private no argument constructor.

**Review by martin-gaievski — APPROVED** [review]
Thank you for addressing all my comments, LGTM

**Review by junqiu-lei — APPROVED** [review]

**Pull request merged**